tests/functional/test_fusion_pg.py

```diff
@@ -0,0 +1,1639 @@
+        {
+            "name": "placement_group1",
+            "tenant": "tenant1",
+            "region": "region1",
+            "availability_zone": "availability_zone1",
+            "storage_service": "storage_service1",
+            "state": "present",
+            "issuer_id": "ABCD1234",
+            "private_key_file": "private-key.pem",
+        },
+        # 'region` is missing
+        {
+            "name": "placement_group1",
+            "tenant": "tenant1",
+            "tenant_space": "tenant_space1",
+            "availability_zone": "availability_zone1",
+            "storage_service": "storage_service1",
+            "state": "present",
+            "issuer_id": "ABCD1234",
+            "private_key_file": "private-key.pem",
+        },
+        # 'availability_zone` is missing
+        {
+            "name": "placement_group1",
+            "tenant": "tenant1",
+            "tenant_space": "tenant_space1",
+            "region": "region1",
+            "storage_service": "storage_service1",
+            "state": "present",
+            "issuer_id": "ABCD1234",
+            "private_key_file": "private-key.pem",
+        },
+        # 'storage_service` is missing
+        {
+            "name": "placement_group1",
+            "tenant": "tenant1",
+            "tenant_space": "tenant_space1",
+            "region": "region1",
+            "availability_zone": "availability_zone1",
+            "state": "present",
+            "issuer_id": "ABCD1234",
+            "private_key_file": "private-key.pem",
+        },
+        # 'state` is invalid
+        {
+            "name": "placement_group1",
+            "tenant": "tenant1",
+            "tenant_space": "tenant_space1",
+            "region": "region1",
+            "availability_zone": "availability_zone1",
+            "storage_service": "storage_service1",
+            "state": "past",
+            "issuer_id": "ABCD1234",
+            "private_key_file": "private-key.pem",
+        },
+    ],
+)
+def test_module_args_wrong(pg_api_init, op_api_init, module_args):
+    set_module_args(module_args)
+
+    pg_mock = MagicMock()
+    pg_mock.get_placement_group = MagicMock(side_effect=purefusion.rest.ApiException)
+    pg_mock.create_placement_group = MagicMock(side_effect=NotImplementedError())
+    pg_mock.update_placement_group = MagicMock(side_effect=NotImplementedError())
+    pg_mock.delete_placement_group = MagicMock(side_effect=NotImplementedError())
+    pg_api_init.return_value = pg_mock
+
+    op_mock = MagicMock()
+    op_mock.get_operation = MagicMock(side_effect=purefusion.rest.ApiException)
+    op_api_init.return_value = op_mock
+
+    with pytest.raises(AnsibleFailJson):
+        fusion_pg.main()
+
+    if pg_mock.get_placement_group.called:
+        pg_mock.get_placement_group.assert_called_with(
+            tenant_name="tenant1",
+            tenant_space_name="tenant_space1",
+            placement_group_name="placement_group1",
+        )
```

Comment:
This applies only for the test cases with `# region is missing`, `# availability_zone is missing` and `# storage_service is missing`, right? What I'm concerned is if the module would call `get_placement_group` before parameters were parsed, this test wouldn't catch it. I think this should be split into two separate tests. WDYT?

Reply:
True. I've actually wanted these lines to rather check whether I did not pass wrong params, e.g. region where tenant should've been, but will add a conditional to do it always where it should